libcaf_net/caf/net/web_socket/framing.cpp

```diff
@@ -267,9 +267,11 @@ class framing_impl : public framing {
     if (hdr_.opcode == detail::rfc6455::text_frame
         || (hdr_.opcode == detail::rfc6455::continuation_frame
             && opcode_ == detail::rfc6455::text_frame))
-      down_->configure_read(receive_policy::up_to(hdr_.payload_len));
+      down_->configure_read(
+        receive_policy::up_to(static_cast<uint32_t>(hdr_.payload_len)));
     else
-      down_->configure_read(receive_policy::exactly(hdr_.payload_len));
+      down_->configure_read(
+        receive_policy::exactly(static_cast<uint32_t>(hdr_.payload_len)));
     return hdr_bytes;
```

Comment:
These casts narrow the `payload_len` field, which is 64-bit according to the RFC. Changing our `receive_policy` to `size_t` solves this, but only on 64-bit architectures. So this needs a `if constexpr (sizeof(size_t) < sizeof(uint64_t))` and some bound checking on platforms where `size_t` is smaller.